### PR TITLE
AoC 2019 Day 2 Part 1

### PR DESCRIPTION
**Pull request by P-on-D**
IntCode! Love this! Takes me back to the old days 😄 

I'm not claiming this as _good_ D code. I'm claiming this as **awesome** D code. You cannot doubt that this is going to work or it is going to crash. Follow the commits.

### CODE REVIEW

**Comment by P-on-D** [comment]
I regret not committing the steps I took to build out `parse` and `execute` but the policy is to commit when the example is complete.

**Comment by P-on-D** [comment]
Because I mentioned it earlier, here's the code on Godbolt in [unoptimised](https://godbolt.org/z/aT6UzE) and [heavily optimised](https://godbolt.org/z/jMQq6x) forms. I don't know if the generated code is awesome or not.

File: 2019/d02p1IntCode.d [hunk 1]
```diff
@@ -78,4 +80,21 @@ unittest {
   parse(8, i);
   assert(!execute(i));
   assert(memory == [30,1,1,4,2,5,6,0,99]);
+} version (unittest) {} else {
```
Comment (P-on-D-Reviewer) [inline 2019/d02p1IntCode.d:83]: What is this?

**Review by P-on-D-Reviewer — COMMENTED** [review]
Looks good to me. Explanatory comments only.

File: 2019/d02p1IntCode.d (hunk 1 above)
Comment (P-on-D) [reply 2019/d02p1IntCode.d:83]: This lets me run the unit tests without the main function using `rdmd -unittest -main` or run the main function if the unit tests pass by omitting `-main`. The [version](https://dlang.org/spec/version.html) feature has so much more but I love it for this alone. It helps me evolve working code.

**Review by P-on-D-Reviewer — APPROVED** [review]

**Pull request merged**